agent/app/internal/agent/pods/agent/agent_test.go

```diff
@@ -2,186 +2,641 @@ package agent
+				v1.PodStatus{},
+			},
+			logs: tests.Logs{
+				Namespace: namespace,
+				Pod:       pod,
+				Container: container,
+				Logs:      logs,
+			},
 		},
-		t,
-	)
+	}
+
+	for _, test := range testCase {
+
+		testFunc := func(t *testing.T) {
+			res := make(chan data.Chunk)
+
+			agent := Agent{
+				kubernetesClient: tests.NewMockKubernetesApiClient(
+					[]string{test.namespace},
+					[]v2.Deployment{},
+					[]v2.StatefulSet{},
+					[]v2.DaemonSet{test.daemonSet},
+					[]v1.Pod{test.pod},
+					[]tests.Logs{test.logs},
+				),
+				results:                     res,
+				readTimestamps:              make(map[string]int64, 0),
+				maxPodPacketSizeBytes:       1000,
+				maxContainerPacketSizeBytes: 1000,
+				clusterId:                   clusterId,
+			}
+
+			go agent.fetchDaemonSetLogsSinceTime(test.namespace, []v2.DaemonSet{test.daemonSet})
+
+			msg := <-res
+
+			assert.Equal(t, msg.ClusterId, clusterId)
+			assert.Equal(t, "\n"+logs, msg.Pods[0].Containers[0].Content)
+		}
+
+		t.Run(test.name, testFunc)
+	}
 }
 
-type containerLogsSplitParams struct {
-	Logs                     string
-	ContainerPacketSizeBytes int
-	ExpectedContainers       int
+func TestFetchNamespaces(t *testing.T) {
+
+	testCase := []struct {
+		name               string
+		namespaces         []string
+		excludedNamespaces []string
+		result             []string
+	}{
+		{
+			name:               "Fetch namespaces with excluded namespaces",
+			namespaces:         []string{"ns-1", "ns-2", "ns-3"},
+			excludedNamespaces: []string{"ns-3", "ns-4"},
+			result:             []string{"ns-1", "ns-2"},
+		},
+		{
+			name:               "Fetch namespaces without excluded namespaces",
+			namespaces:         []string{"ns-1", "ns-2", "ns-3"},
+			excludedNamespaces: []string{},
+			result:             []string{"ns-1", "ns-2", "ns-3"},
+		},
+		{
+			name:               "Fetch empty namespaces",
+			namespaces:         []string{},
+			excludedNamespaces: []string{},
+			result:             []string{},
+		},
+		{
+			name:               "Fetch namespaces without overlapping excluded namespaces",
+			namespaces:         []string{"ns-1", "ns-2"},
+			excludedNamespaces: []string{"ns-3", "ns-4"},
+			result:             []string{"ns-1", "ns-2"},
+		},
+	}
+
+	for _, test := range testCase {
+
+		testFunc := func(t *testing.T) {
+			agent := Agent{
+				excludedNamespaces: test.excludedNamespaces,
+				kubernetesClient: tests.NewMockKubernetesApiClient(
+					test.namespaces,
+					[]v2.Deployment{},
+					[]v2.StatefulSet{},
+					[]v2.DaemonSet{},
+					[]v1.Pod{},
+					[]tests.Logs{},
+				),
+			}
+
+			agent.fetchNamespaces()
+
+			res := agent.includedNamespaces
+			assert.Equal(t, res, test.namespaces)
+		}
+
+		t.Run(test.name, testFunc)
+	}
+
 }
 
-func testContainerLogsPacketSplit(params containerLogsSplitParams, t *testing.T) {
-	agent := Agent{maxContainerPacketSizeBytes: params.ContainerPacketSizeBytes}
+func TestGatherClusterMetadata(t *testing.T) {
+
```

Comment:
The most common convention for naming tests is Test<StructName>_MethodName

Reply:
Or just TestMethodName if it doesn't have a receiver

Reply:
Actually this test follows the rule :D It tests the "GatherClusterMetadata" method.